src/core/main.c

```diff
@@ -1338,7 +1338,8 @@ int main(int argc, char *argv[]) {
         saved_argv = argv;
         saved_argc = argc;
 
-        log_show_color(isatty(STDERR_FILENO) > 0);
+        log_show_color(!streq_ptr(getenv("TERM"), "dumb") &&
+                       isatty(STDERR_FILENO) > 0);
```

Comment:
The semantic are slightly different: colors_enabled checks stdout, this checks stderr, and colors_enabled checks $SYSTEMD_COLORS, but this doesn't. But yeah, maybe this distinction doesn't matter and we should also look at $SYSTEMD_COLORS here. I'll change it to call colors_enabled().